### PR TITLE
moved metric class from agent to sdk

### PR DESCRIPTION
**Pull request by gabriellewp**
Signed-off-by: Gabrielle Poerwawinata <gabrielle.w.poerwawinata@intel.com>

### CODE REVIEW

**Comment by srware** [comment]
@gabriellewp , this seems sensible. We need to update the agent in-line with this change also.

**Comment by gabriellewp** [comment]
Hi Scott, 

Question.. is that would be fine to aligned Metric.data.js on the same level of hierarchy as common.js, httpCllient.js etc files? So in that case I dont need to put Metric.data.js inside of data folder. I suggest this because when I call module.data = require('./data'); in index.js, it gets linked to data function in lib/proxies/iot.mqtt.js instead of data folder where I put metric class inside of it.

Gaby

**Comment by wagmarcel** [comment]
@srware on motivation: The MQTT Gateway is using  this library to decode the MQTT Payload. So you have to know this library to encode payload for MQTT. This mandates it to be part of the SDK rather than of the agent IMHO. Otherwise no one can send MQTT witthout the Agent and e.g. the E2E test has either to use the Agent(which would be odd) or using their own version of this library (which would be bad design, too).
This is in general not a great design, it has historic reasons.
Mid-term plan is to align MQTT payload with data-sending REST API, so that everyone can create payload according to the spec. However, this will need some work and will not part of the AA release.

**Comment by srware** [comment]
@wagmarcel, thanks for the explanation, this is more or less what I expected.

@gabriellewp, I don't like the refactoring. Having a 'Metric' directory doesn't make much sense. I would assume the issues you were encountering with the 'data/metric.data.js' in because there is no 'index.js' in the 'data' directory so nothing is getting exported...

My proposal would be refactor to 'lib/data/metric.js' (all lowercase!!!), update the index.js in 'lib' to expose './data' and add an index.js to 'lib/data' to expose 'metric.js' similar to [this](https://github.com/Open-IoT-Service-Platform/oisp-sdk-js/blob/development/api/mqtt/index.js).

Thanks

**Comment by srware** [comment]
Also, please make sure the commit message makes sense for what is being committed. The current PR commit says "expose metric to the index.js" as the sole commit (I assumed this has been squashed) but it doesn't actually describe what the PR is doing.

Something like "Include metric class in SDK" is fine.

Thanks

File: lib/index.js [hunk 1]
```diff
@@ -26,7 +26,8 @@ SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 
 module.exports = function(config) {
     var module = {};
-    
+
+    module.data = require('./data');
```
Comment (srware) [inline lib/index.js]: For my OCD can you make the "= require" align with the others :)

Comment (gabriellewp) [reply lib/index.js]: How about now?

File: lib/data/metric.js [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+    delete this.did;
+    delete this.count;
+    delete this.nameOfComponentId;
+    delete this.globalCid;
+    return this;
+};
+
+module.exports.init = function () {
+    return Metric;
+
```
Comment (srware) [inline lib/data/metric.js]: Can you please remove this additional whitespace.

**Comment by srware** [comment]
Something has been force pushed/merged with conflicts and committed leaving all of the automated comments in the code... When this is sorted please squash this down to a single commit and make sure the additional description in the commit message is on a new line. The first line of the commit message should simply be "Include Metric in the SDK".

Thanks

File: lib/data/metric.js (hunk 2 above)
Comment (gabriellewp) [reply lib/data/metric.js]: done..

**Pull request merged**